### PR TITLE
Setup pagination for transactions and invoices

### PR DESCRIPTION
**Pull request by zulandar**
Setup pagination on main page. Also setup CSS on this as well. Below is the screen shot on recent transactions. Current limit is 5 per page.

Transaction_Pagination

Invoice:
Invoice_Pagination

@SonarSoftware we can go through a live demo on this design if you want to.

### CODE REVIEW

**Comment by zulandar** [comment]
UX design assistance from @matbram

File: app/Http/Controllers/BillingController.php [hunk 1]
```diff
@@ -662,4 +667,29 @@ private function formatHistoricalUsageData($historicalUsageData)
...
+            $page = 1; // Get the current page or default to 1
+        }
+
+
+        $items = $items instanceof Collection ? $items : Collection::make($items);
+
+        return new LengthAwarePaginator($items->forPage($page, $perPage), $items->count(), $perPage, $page, $options);
+    }
+
+    private function cleanUrl($url)
```
Comment (SonarSoftware) [inline app/Http/Controllers/BillingController.php]: Why are you replacing https:// URLs with http:// ?

File: app/Http/Controllers/BillingController.php [hunk 2]
```diff
@@ -662,4 +667,29 @@ private function formatHistoricalUsageData($historicalUsageData)
...
 
+    private function paginate($items, $perPage, $setDefaultOption = true, $options = [])
+    {
+        if($setDefaultOption){
+            $options = ['path' => request()->url(), 'query' => request()->query()];
+        }
+
+        $requestUrl = $this->cleanUrl(request()->url());
+
+        if (isset($options['path']) && env('NGINX_HOST') . $options['path'] == $requestUrl) {
```
Comment (SonarSoftware) [inline app/Http/Controllers/BillingController.php]: This seems a bit weird.. can we not use something like `{$_SERVER['HTTP_HOST']}{$_SERVER['REQUEST_URI']}` ?

File: app/Http/Controllers/BillingController.php (hunk 1 above)
Comment (zulandar) [reply app/Http/Controllers/BillingController.php]: This is actually a one line to replace `https://` and `http://`. Tested and this logic works in regards to remove both `https://` and `http://`

File: app/Http/Controllers/BillingController.php (hunk 2 above)
Comment (zulandar) [reply app/Http/Controllers/BillingController.php]: I could do that but then would have to remove the page off the end.

Here is the output for the three different ways.

`$_SERVER['HTTP_HOST']` = "portal.test"
`$_SERVER['REQUEST_URI']` = "/portal/billing/invoices?page=3"

`$requestUrl = $this->cleanUrl(request()->url());`
`$requestUrl` = "portal.test/portal/billing/invoices"

Comment (SonarSoftware) [reply app/Http/Controllers/BillingController.php]: Yeah, it was more grabbing the NGINX_HOST from env seems weird to me. What if someone forked this code and wasn't running it on NGINX anymore? Doesn't it make more sense to just use the basic PHP $_SERVER variables?

Comment (zulandar) [reply app/Http/Controllers/BillingController.php]: Makes sense. I've adjusted it in the last commit to pull from  `$_SERVER['HTTP_POST']`

**Pull request merged**